frontend/src/shared/helpers.tsx

```diff
@@ -140,18 +140,69 @@ export const RequestsListColumns: ColDef[] = [
   },
 ];
 
+export function preparePatientDataForAgGrid(
+  patientsListQueryResult: PatientsListQuery
+) {
+  const { patientsConnection, patients } = patientsListQueryResult;
+
+  const newPatients = patients.map((p) => {
+    const {
+      smilePatientId,
+      patientAliasesIsAlias: patientAliases,
+      hasSampleSamplesConnection: samplesConnection,
+      hasSampleSamples: samples,
+    } = p;
+    const cmoPatientId = patientAliases?.find(
+      (pa) => pa.namespace === "cmoId"
+    )?.value;
+    const dmpPatientId = patientAliases?.find(
+      (pa) => pa.namespace === "dmpId"
+    )?.value;
+
+    const totalSamples = samplesConnection?.totalCount;
+
+    const cmoSampleIds = samples?.map((s) => {
+      const sampleMetadata = s.hasMetadataSampleMetadata[0];
+      return sampleMetadata?.cmoSampleName || sampleMetadata?.primaryId;
+    });
+
+    const additionalProperties =
+      samples[0]?.hasMetadataSampleMetadata[0]?.additionalProperties;
+    const additionalPropertiesJson = additionalProperties
+      ? JSON.parse(additionalProperties)
+      : {};
+    const consentPartA = additionalPropertiesJson["consent-parta"];
+    const consentPartC = additionalPropertiesJson["consent-partc"];
+
+    return {
+      cmoPatientId,
+      dmpPatientId,
+      totalSamples,
+      cmoSampleIds,
+      consentPartA,
+      consentPartC,
+      smilePatientId,
+    };
+  });
+
+  return {
+    patientsConnection,
+    patients: newPatients,
+  };
+}
+
```

Comment:
See [this comment above](https://github.com/mskcc/smile-dashboard/pull/119/files#r1576559456).